mmtk/Cargo.toml

```diff
@@ -44,17 +44,18 @@ memoffset = "*"
 # ykstackmaps = { git = "https://github.com/udesou/ykstackmaps.git", branch = "udesou-master", version = "*" }
 
 [features]
-default = ["mmtk/vm_space", "julia_copy_stack", "object_pinning", "is_mmtk_object"]
+default = ["mmtk/vm_space", "julia_copy_stack"]
 
 # Plans
 nogc = []
 immix = []
 stickyimmix = ["mmtk/sticky_immix_non_moving_nursery", "mmtk/immix_smaller_block"]
 marksweep = []
-object_pinning = ["mmtk/object_pinning"]
 
 # This feature disables moving
 non_moving = ["mmtk/immix_non_moving", "mmtk/immix_smaller_block"]
-julia_copy_stack = []
-# Feature necessary to support conservative stack scanning
+object_pinning = ["mmtk/object_pinning"]
 is_mmtk_object = ["mmtk/is_mmtk_object"]
+conservative = ["is_mmtk_object"]
+moving = ["conservative", "object_pinning"]
```

Comment:
Should moving be in the `default`? Or `non_moving`? 
Ideally, people can build with`cargo build --features immix/stickyimmix` and then get immix/sticky working with or without moving, whatever the default is. But as it is, I think we get moving but without conservative stack scanning which tpins everything I assume? That being said, maybe we should have non-moving as default?

Reply:
We can put it to default. But the default features will be included in all the builds, unless we pass `--no-default-features`. So when we build the non moving build, we will need to do `cargo build --no-default-features --features non_moving,mmtk/vm_space,julia_copy_stack`.

Reply:
I can revert his change to what we had before if you think that is better.

Reply:
The reason that I changed this is that I found the following line confusing: `object_pinning` is turned on even for a non moving build.
`#[cfg(all(feature = "object_pinning", not(feature = "non_moving")))]`

When the changes in this PR, we will need `conservative` as a default feature to support moving. We will need to do the same thing for `conservative`. 

I will see if there is a better way.

Reply:
If it's too difficult, it's also fine to leave as it is. The other thing to have in mind is that we should probably update the README file to reflect the changes - reminding people to set `MMTK_CONSERVATIVE=1` for example.

Reply:
I mostly revert back to what we had before. But I removed some features from conditional compilation in the binding crate, like `is_mmtk_object`, `object_pinning`. So we always build with those features, as they are default features -- we have to build with default features.

On top of default features, people may build with `non_moving`. With `non_moving`, some methods will do early return and skip the actual work. Like pinning, or conservative scanning.